dump/dbus_util.cpp

```diff
@@ -83,62 +83,49 @@ uint64_t getDumpSize(sdbusplus::bus::bus& bus, const std::string& objectPath)
 
 bool isSystemHMCManaged(sdbusplus::bus::bus& bus)
 {
-    using BiosBaseTableItem = std::pair<
-        std::string,
-        std::tuple<std::string, bool, std::string, std::string, std::string,
-                   std::variant<int64_t, std::string>,
-                   std::variant<int64_t, std::string>,
-                   std::vector<std::tuple<
-                       std::string, std::variant<int64_t, std::string>>>>>;
-    using BiosBaseTable = std::vector<BiosBaseTableItem>;
-
     try
     {
-        std::string hmcManaged{};
-        auto retVal = readDBusProperty<std::variant<BiosBaseTable>>(
+        auto retVal = readDBusProperty<std::variant<BaseBIOSTableItemList>>(
             bus, "xyz.openbmc_project.BIOSConfigManager",
             "/xyz/openbmc_project/bios_config/manager",
             "xyz.openbmc_project.BIOSConfig.Manager", "BaseBIOSTable");
-        const auto baseBiosTable = std::get_if<BiosBaseTable>(&retVal);
-        if (baseBiosTable == nullptr)
-        {
-            log<level::ERR>(
-                "Util failed to read BIOSconfig property BaseBIOSTable");
-            return false;
-        }
-        for (const auto& item : *baseBiosTable)
+
+        if (std::holds_alternative<BaseBIOSTableItemList>(retVal))
         {
-            std::string attributeName = std::get<0>(item);
-            auto attrValue = std::get<5>(std::get<1>(item));
-            auto val = std::get_if<std::string>(&attrValue);
-            if (val != nullptr && attributeName == "pvm_hmc_managed")
+            const auto& baseBiosTable = std::get<BaseBIOSTableItemList>(retVal);
+            auto it = baseBiosTable.find("pvm_hmc_managed");
+            if (it != baseBiosTable.end())
             {
-                hmcManaged = *val;
-                break;
+                const auto& attrValue = std::get<5>(it->second);
+                if (std::holds_alternative<std::string>(attrValue))
+                {
+                    const std::string& strValue =
+                        std::get<std::string>(attrValue);
+                    log<level::INFO>(
+                        fmt::format("isSystemHMCManaged value ({})", strValue)
+                            .c_str());
+                    return strValue == "Enabled";
+                }
+                else
+                {
+                    log<level::ERR>(
+                        "Unexpected value type for 'pvm_hmc_managed");
+                    return false;
+                }
+            }
+            else
+            {
```

Comment:
Suggestion: I feel there are around 3 else conditions here. I know this helps in identifying the type of error. But if that is not really needed to that level, we could just have one error and return false. I am not sure, how often it could fail and the how critical it is to have a log trace describing the exact issue. You could take a call